### ISSUE TITLE
Pickling fields

### ISSUE THREAD

**Issue by bpopeters** [OpenNMT/OpenNMT-py#909]
I have seen two main reasons why pickling `torchtext.data.Field` objects is a problem:

1. Lambdas are not serializable.
2. `torch.dtype` is not serializable.

The first issue should not matter because any lambda can be replaced by a function. Functions _are_ serializable. The default `tokenize` argument in `torchtext.data.Field` uses a spurious lambda, but the problem goes away if you use a serializable tokenizer instead.

The second issue is fixed in 0.4.1. I recently updated pytorch and I can confirm that fields can be saved and loaded with `torch.save` and `torch.load`. Right now, there is a lot of code in master related to saving and loading fields and vocabs that clutters up what should be a pretty simple I/O process and makes it difficult to find bugs and inefficiencies.

Is there interest in trying to make use of the saveability of fields? The upside is that the I/O code should get quite a bit simpler and shorter and the downside is that it makes it difficult to support older versions of pytorch. I would argue that having a maintainable and extensible codebase is more important than supporting old versions.

**Comment by vince62s** [comment]
I agree with your last commment.
It is just that making the code 0.4.1 compatible first has to happen before. I never tried on my end so far.

**Comment by bpopeters** [comment]
As far as I can tell, the changes in 0.4.1 were minor: the only issue I've come across is the thing with the type of `torch.arange`, and there's already a pull request to take care of it.

**Comment by vince62s** [comment]
okay, I was under the impression that there was another issue, when you go through this thread. https://github.com/OpenNMT/OpenNMT-py/issues/887

**Comment by bpopeters** [comment]
Hmm, thanks. I wasn't aware of that one. My officemates and I have trained on multiple GPUs using 0.4.1 without running into that issue (albeit not with the transformer), so I'm not sure I can shed any light.

**Comment by vince62s** [comment]
Ben I find it weird you did not have the issue in multi-gpu training with 0.4.1
I just tried it and it does trigger an error. see #921
not just the transformer, any model.
